docs_src/microservices/application/V3Migration.md

```diff
@@ -1,336 +1,231 @@
+    4. Remove remaining **Trigger** configuration
 
-!!! example "Example - External MQTT trigger configuration"
+#### External MQTT changes
 
-    ```toml
-    [Trigger]
-    Type="external-mqtt"
-      [Trigger.ExternalMqtt]
-      Url = "tcp://broker.hivemq.com:1883"
-      SubscribeTopics = "edgex-trigger"
-      PublishTopic = "edgex-trigger-response"
-      ClientId = "app-my-service"
-      ConnectTimeout = "30s"
-      AutoReconnect = false
-      KeepAlive = 60
-      QoS = 0
-      Retain = false
-      SkipCertVerify = false
-      SecretPath = ""
-      AuthMode = "none"
+The **External MQTT** trigger configuration remains under **Trigger** configuration, but the `SubscribeTopics` and `PublishTopic` setting have been moved to the top level of the **Trigger** configuration. 
+
+!!! example "Example - External MQTT trigger configuration"
+    ```yaml
+    Trigger:
+      Type: "external-mqtt"
+      SubscribeTopics: "external-request/#"
+      PublishTopic: "" # optional if publishing response back to the the External MQTT Broker
+      ExternalMqtt:
+        Url: "tcp://broker.hivemq.com:1883" #  fully qualified URL to connect to the MQTT broker
+        ClientId: "app-my-service"
+        ConnectTimeout: "30s" 
+        AutoReconnect: true
+        KeepAlive: 10 # Seconds (must be 2 or greater)
+        QoS: 0 # Quality of Service 0 (At most once), 1 (At least once) or 2 (Exactly once)
+        Retain: true
+        SkipCertVerify: false
+        SecretName: "mqtt-trigger" 
+        AuthMode: "none"
     ```
 
-##### HTTP
+####  external-mqtt **Trigger** Migration
 
-The HTTP trigger configuration has not changed beyond the renaming of `Binding` to `Trigger`.
+1. Move your `SubscribeTopics` and `PublishTopic` top level of the **Trigger** configuration
 
-!!! example "Example - HTTP trigger configuration"
+#### HTTP Changes
 
-    ```toml
-    [Trigger]
-    Type="http"
-    ```
+The HTTP trigger configuration has not changed for EdgeX 3.0
+
+### Writable Pipeline
+
+See [Pipeline Configuration](#pipeline-configuration) section below for changes to the Writable Pipeline configuration
+
+## Custom Application Service 
 
 ### Code
 
 #### Dependencies
 
-You first need to update the `go.mod` file to specify `go 1.16` and the V2 versions of the App Functions SDK and any EdgeX go-mods directly used by your service. Note the extra `/v2` for the modules.
+You first need to update the `go.mod` file to specify `go 1.20` and the v3 versions of the App Functions SDK and any EdgeX go-mods directly used by your service. 
 
-!!! example "Example go.mod for V2"
+!!! example "Example go.mod for V3"
 
     ```go
     module <your service>
     
-    go 1.16
+    go 1.20
     
     require (
-    	github.com/edgexfoundry/app-functions-sdk-go/v2 v2.0.0
-    	github.com/edgexfoundry/go-mod-core-contracts/v2 v2.0.0
+    	github.com/edgexfoundry/app-functions-sdk-go/v3 v3.0.0
+    	github.com/edgexfoundry/go-mod-core-contracts/v3 v3.0.0
     )
     ```
 
```

Comment:
Do we need to add a note about running `make tidy` or `go mod tidy` to update the rest of the go.mod/go.sum files? or is that implied?